### PR TITLE
[config] Configurable MLE upgrade/downgrade defaults

### PR DESCRIPTION
**Pull request by pdgendt**
Add more MLE configuration definitions which can be overriden at build time

### CODE REVIEW

File: src/core/config/mle.h [hunk 1]
```diff
@@ -48,6 +48,36 @@
...
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_MLE_MIN_DOWNGRADE_NEIGHBORS
+ *
+ * The minimum of downgrade neighbors.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_MLE_MIN_DOWNGRADE_NEIGHBORS
+#define OPENTHREAD_CONFIG_MLE_MIN_DOWNGRADE_NEIGHBORS 7
```
Comment (jwhui) [inline src/core/config/mle.h:58]: @pdgendt , thanks for submitting this PR.

Can you describe your use case for making these build time configurable?

Note that these values are specified in the Thread Protocol Specification and changing these values will result in a non-conformant implementation.

Comment (pdgendt) [reply src/core/config/mle.h:58]: @jwhui `OPENTHREAD_CONFIG_MLE_MIN_DOWNGRADE_NEIGHBORS` can be left out if that's an issue, I added it for consistency regarding the downgrade/upgrade variables.

For the router downgrade/upgrade threshold values a user is able to modify these at runtime, so it made sense to have configurable default values.

The use case is to have a REED prefer the role of router, as far I could find this was the only available setting that has impact as you can't force a role, unless there is an alternative way?

Comment (pdgendt) [reply src/core/config/mle.h]: @jwhui I've removed `OPENTHREAD_CONFIG_MLE_MIN_DOWNGRADE_NEIGHBORS` from the PR, are the default threshold definition changes ok?

Comment (jwhui) [reply src/core/config/mle.h]: > For the router downgrade/upgrade threshold values a user is able to modify these at runtime, so it made sense to have configurable default values.

The Thread Specification specifies values for both the upgrade and downgrade thresholds. The reason OpenThread has CLI commands to change those values is to support testing. See the note in the API docs:
https://github.com/openthread/openthread/blob/34366834b04cd168053edc224c5854d5b20dd769/include/openthread/thread_ftd.h#L347-L348

> The use case is to have a REED prefer the role of router, as far as I could find this was the only available setting that has impact as you can't force a role, unless there is an alternative way?

Unfortunately, the Thread 1.1.1 Specification does not currently offer a way to prefer a given REED to become a router.

Comment (pdgendt) [reply src/core/config/mle.h]: Ok, thanks for the explanation.

**Pull request closed without merging**